src/useCases/UploadToS3UseCase.ts

```diff
@@ -10,7 +11,7 @@ import { AmzDocumentKey } from '../generated/definitions/AmzDocumentKey';
 export const UploadToS3UseCase =
   (uploadToS3Repository: UploadToS3RecordRepository) =>
   (key: AmzDocumentKey) =>
-  (checksumAlg: AmzSdkChecksumAlg) =>
+  (checksumAlg: O.Option<AmzSdkChecksumAlg>) =>
```

Comment:
try to do not take monads as input (it's caller duty to do the checks _before_ passing it to the method)